Modules/RoleRequestModule.cs

```diff
@@ -0,0 +1,127 @@
+using System.Text;
+using Discord;
+using Discord.Commands;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using GeneralPurposeBot.Services;
+using GeneralPurposeBot.Models;
+
+namespace DiscordBot.Modules
+{
+
+    [Group("!role"), Summary("!role set <roleName> to set your role, !role remove to remove your role")]
+    [RequireBotPermission(Discord.GuildPermission.ManageRoles, ErrorMessage = "The bot must be able to manage roles on the server to set user roles.", NotAGuildErrorMessage = "This must be run inside of a server.")]
+    public class RoleRequestModule : ModuleBase
+    {
+        public RoleRequestService RoleRequestService { get; set; }
+        public RoleRequestModule(RoleRequestService roleRequestService) { RoleRequestService = roleRequestService; }
+
+        [Command("add"), Summary("!role add <roleName> to add the role to the list of self-assignable roles. Role must not have moderation-related permissions to work.")]
+        [RequireUserPermission(GuildPermission.ManageRoles, ErrorMessage = "You must be able to manage roles to set a role as self-assignable.")]
+        public async Task Add(string role)
+        {
+            var possibleRoles = Context.Guild.Roles.Where(r => r.Name == role);
+            if (!possibleRoles.Any())
+            {
+                await Context.Channel.SendMessageAsync("Role does not exist!");
+            }
+            else
+            {
+                var searchRole = possibleRoles.First();
+                if (RoleRequestService.GetRole(searchRole.Id) == null)
+                {
+                    if (!searchRole.Permissions.ManageRoles)
+                    {
+                        var newRole = new AssignableRole()
+                        {
+                            RoleId = searchRole.Id,
+                            ServerId = Context.Guild.Id,
+                            RoleName = searchRole.Name
+                        };
+                        RoleRequestService.AddRole(newRole);
+                    }
+                    else
+                        await Context.Channel.SendMessageAsync("Moderated roles can't be self-assignable!");
```

Comment:
Keep braces consistent between if and else
```suggestion
                    else 
                    {
                        await Channel.Context.SendMessageAsync("Moderated roles can't be self-assignable!");
                    }
```